Emergence/Contexts/Networking/Observable+Networking.swift

```diff
@@ -0,0 +1,49 @@
+import RxSwift
+import Gloss
+import Moya
+
+enum ORMError : ErrorType {
+    case ORMNoRepresentor
+    case ORMNotSuccessfulHTTP
+    case ORMNoData
+    case ORMCouldNotMakeObjectError
+}
+
+extension Observable {
+    func mapSuccessfulHTTPToObject(classType: Decodable.Type) -> Observable<Decodable> {
+
+        func resultFromJSON(object:[String: AnyObject], classType: Decodable.Type) -> Decodable? {
+            return classType.init(json: object)
+        }
+
+        return map { representor in
+            guard let response = representor as? MoyaResponse else {
+                throw ORMError.ORMNoRepresentor
+            }
+
+            // Allow successful HTTP codes
+            if ((200...209) ~= response.statusCode) == false {
```

Comment:
this can be done with a `guard`:

guard ((200...209) ~= response.statusCode) else {
    if let json = try? NSJSONSerialization.JSONObjectWithData(response.data, options: .AllowFragments) as? [String: AnyObject] {
        print("Got error message: \(json)")
    }
    throw ORMError.ORMNotSuccessfulHTTP
}